### PR TITLE
Bug 543398 - Proposal for new Widget: Password Revealer

### PR DESCRIPTION
**Pull request by lcaron**
Initial Release

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=543398

File: widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java [hunk 1]
```diff
@@ -0,0 +1,2357 @@
...
+ * <p>
+ * IMPORTANT: This class is <em>not</em> intended to be subclassed.
+ * </p>
+ *
+ * @see <a href="http://www.eclipse.org/swt/snippets/#text">Text snippets</a>
+ * @see <a href="http://www.eclipse.org/swt/examples.php">SWT Example: ControlExample</a>
+ * @see <a href="http://www.eclipse.org/swt/">Sample code and further information</a>
+ * @noextend This class is not intended to be subclassed by clients.
+ */
+public class PasswordRevealer extends Composite {
```
Comment (mickaelistria) [inline widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java:80]: Have you considered making it implement Text instead?

**Review by mickaelistria — COMMENTED** [review]
Cou

File: widgets/roundedcheckbox/org.eclipse.nebula.widgets.roundedcheckbox.snippets/src/org/eclipse/nebula/widgets/roundedcheckbox/snippets/MouseNavigator.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
+/*******************************************************************************
```
Comment (wimjongman) [inline widgets/roundedcheckbox/org.eclipse.nebula.widgets.roundedcheckbox.snippets/src/org/eclipse/nebula/widgets/roundedcheckbox/snippets/MouseNavigator.java]: Did you add this by accident?

File: widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java [hunk 3]
```diff
@@ -0,0 +1,2357 @@
...
+		textField.setLayoutData(gdText);
+		textField.setVisible(false);
+		gdText.exclude = true;
+
+		eyeButton = new EyeButton(this, SWT.NONE);
+		eyeButton.setLayoutData(new GridData(GridData.FILL, GridData.FILL, false, false));
+
+		passwordField.addListener(SWT.Modify, e -> {
+			final String currentText = passwordField.getText();
+			textField.setText(currentText);
```
Comment (wimjongman) [inline widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java]: Is it an idea to only field textField when the eye button is pressed and then clear the textField when eye is depressed?

File: widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java (hunk 1 above)
Comment (lcaron) [reply widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java:80]: I thought about that but the main idea is to have 2 widgets, one visible (password) and the other one unvisible, so I needed a composite

File: widgets/roundedcheckbox/org.eclipse.nebula.widgets.roundedcheckbox.snippets/src/org/eclipse/nebula/widgets/roundedcheckbox/snippets/MouseNavigator.java (hunk 2 above)
Comment (lcaron) [reply widgets/roundedcheckbox/org.eclipse.nebula.widgets.roundedcheckbox.snippets/src/org/eclipse/nebula/widgets/roundedcheckbox/snippets/MouseNavigator.java]: Oooppss... "Commit all" button side effect.

File: widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java (hunk 3 above)
Comment (lcaron) [reply widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java]: Excellent idea.

**Comment by sibeon** [comment]
Hi, I know it is a bit late, but I just found that widget tha I just also implemented. I used the possibility to clear the echo character. So I do not have to handle synchronizatio and hiding text widgets.

Is there a reason why you implemented that with two text fields?

**Comment by lcaron** [comment]
Hi Stefan,

Well I did that because I though it was the better implementation.
Can you share this widget please ? I'm very interested and I'll really glad to have a look.

Laurent

**Comment by sibeon** [comment]
[Password.txt](https://github.com/eclipse/nebula/files/4134926/Password.txt)
This is my implementation

**Comment by lcaron** [comment]
Simple and clever ;)
If you don't mind, I'll modify my widget to use this trick. Are you OK with it ?

**Comment by sibeon** [comment]
Shure go ahead.
Would be cool If you would also include a possibility to change the Icon.

**Pull request merged**